lib/elixir/lib/kernel.ex

```diff
@@ -2292,8 +2292,18 @@ defmodule Kernel do
   In order to compare more than two clauses, the `cond/1` macro has to be used.
   """
   defmacro if(condition, clauses) do
-    do_clause = Keyword.get(clauses, :do, nil)
-    else_clause = Keyword.get(clauses, :else, nil)
+    {do_clause, clauses} = Keyword.pop_first(clauses, :do)
+    {else_clause, clauses} = Keyword.pop_first(clauses, :else)
+
+    case {do_clause, else_clause, clauses} do
+      {_do, _else, [{:do, _val} | _tail]} ->
+        raise ArgumentError,
+        "duplicate `do` key in if"
```

Comment:
Need a similar clause for `else` also otherwise the error message will be weird on duplicate `else`.